pkg/controller/syncset/syncset_controller.go

```diff
@@ -566,6 +622,32 @@ func (r *ReconcileSyncSet) reconcileDeletedSyncSetResources(applyMode hivev1.Syn
 	return newStatusList, nil
 }
 
+func (r *ReconcileSyncSet) deleteSyncSetResources(syncSetStatus hivev1.SyncSetObjectStatus, dynamicClient dynamic.Interface, ssLog log.FieldLogger) (hivev1.SyncSetObjectStatus, error) {
+	for index, resourceStatus := range syncSetStatus.Resources {
+		itemLog := ssLog.WithField("resource", fmt.Sprintf("%s/%s", resourceStatus.Namespace, resourceStatus.Name)).
+			WithField("apiversion", resourceStatus.APIVersion).
+			WithField("kind", resourceStatus.Kind)
+		gv, err := schema.ParseGroupVersion(resourceStatus.APIVersion)
+		if err != nil {
+			syncSetStatus.Resources[index].Conditions = r.setDeletionFailedSyncCondition(syncSetStatus.Resources[index].Conditions, err)
```

Comment:
you should likely create a new error indicating that you failed to parse the groupversion, otherwise it may not be straightforward to tell what actually went wrong.